### PR TITLE
add retry when loss is NaN in train and finetune

### PR DESCRIPTION
**Pull request by rayrayraykk**
add retry when loss is NaN in train and finetune when `cfg.llm.retry_on_nan_loss = True`

### CODE REVIEW

**Review by qbc2016 — APPROVED** [review]
LGTM.
